### ISSUE TITLE
Progression Labels on the Top Should Reflect the Filtering

### ISSUE THREAD

**Issue by TigerPurrer** [SeanPrashad/leetcode-patterns#88]
I think progression labels on the top (e.g. Total: 75/170, Easy: 14/36, Medium: 53/102, Hard: 8/32) should be reflecting the list that's present on the screen. This is not something important, but just realized that they were not changing and felt like mentioning it :)

**Comment by SeanPrashad** [comment]
Good catch @TigerPurrer - it makes sense that the `X/Y easy` etc should update when filtering by `Difficulty`, `Pattern`, or a combination of the two. On first thought, I think this will be trickier than it seems but it's still worthwhile to investigate!
